api/groups/eventsData.go

```diff
@@ -30,7 +30,17 @@ func UnmarshallBlockData(marshaller marshal.Marshalizer, marshalledData []byte)
 		return nil, err
 	}
 
-	header, err := unmarshal.GetHeaderFromBytes(marshaller, core.HeaderType(argsBlockS.BlockData.HeaderType), argsBlockS.BlockData.HeaderBytes)
+	err = checkBlockDataValid(argsBlockS)
+	if err != nil {
+		return nil, err
+	}
+
+	headerCreator, err := getHeaderCreator(argsBlockS.BlockData.HeaderType)
```

Comment:
I would refactor this function. 
Instead of having it exported and require that we prepare this container of empty headers creators, why not create a proper struct that will initialize the container once, on the constructor?

Reply:
added a separate component

Reply:
will come with a next PR for cleanup and handling multiple versions for unmarshalling